### PR TITLE
SEC-19137 Add additional get_aws_credentials tests to `spark-run`

### PR DESCRIPTION
**Pull request by choww**
❗️Must be merged after https://github.com/Yelp/paasta/pull/3901

Add tests to spark-run to make sure `get_aws_credentials()` from `service_configuration_lib` is invoked properly by the various flags passed to `paasta spark-run`: 
- `--service` should not overwrite `--aws-profile` 
- use `--service` if `--aws-profile` is absent
- in absence of a valid `--service` and `--aws-profile`, use `--aws-profile=default`

### CODE REVIEW

File: tests/cli/test_cmds_spark_run.py [hunk 1]
```diff
@@ -1450,3 +1450,75 @@ def test_get_aws_credentials():
         RoleSessionName=mock.ANY,
         WebIdentityToken="token-content",
     )
+
+
+@mock.patch("service_configuration_lib.spark_config.use_aws_profile", autospec=False)
+@mock.patch("service_configuration_lib.spark_config.Session", autospec=True)
+def test_get_aws_credentials_session(mock_boto3_session, mock_use_aws_profile):
+    # prioritize session over `profile_name` if both are provided
+    from service_configuration_lib.spark_config import get_aws_credentials
```
Comment (CaptainSame) [inline tests/cli/test_cmds_spark_run.py:1459]: Any reason to not have imports at the top? LGTM otherwise

Comment (choww) [reply tests/cli/test_cmds_spark_run.py]: Moved all the import statements!

File: tests/cli/test_cmds_spark_run.py [hunk 2]
```diff
@@ -1450,3 +1450,64 @@ def test_get_aws_credentials():
...
+
+@mock.patch("service_configuration_lib.spark_config.use_aws_profile", autospec=False)
+@mock.patch("service_configuration_lib.spark_config.Session", autospec=True)
+def test_get_aws_credentials_default_profile(mock_boto3_session, mock_use_aws_profile):
+    # use `default` profile if no valid options are provided
+    get_aws_credentials(
+        service="spark",
+    )
+
+    mock_use_aws_profile.assert_called_once()
```
Comment (CaptainSame) [inline tests/cli/test_cmds_spark_run.py:1513]:
```suggestion
    mock_use_aws_profile.assert_called_once_with()
```

**Review by CaptainSame — APPROVED** [review]
Nit

**Review by Qmando — APPROVED** [review]

**Pull request merged**